### PR TITLE
PEP 660: Editable installs for PEP 517 style build backends

### PR DESCRIPTION
**Pull request by sbidoul**

### CODE REVIEW

**Comment by the-knights-who-say-ni** [comment]
Hello, and thanks for your contribution!

I'm a bot set up to make sure that the project can legally accept this contribution by verifying everyone involved has signed the [PSF contributor agreement](https://www.python.org/psf/contrib/contrib-form/) (CLA).



## CLA Missing

Our records indicate the following people have not signed the CLA:

@dholth

For legal reasons we need all the people listed to sign the CLA before we can look at your contribution. Please follow [the steps outlined in the CPython devguide](https://devguide.python.org/pullrequest/#licensing) to rectify this issue.

If you have recently signed the CLA, please wait at least one business day
before our records are updated.


You can [check yourself](https://check-python-cla.herokuapp.com/) to see if the CLA has been received.

Thanks again for the contribution, we look forward to reviewing it!

**Comment by sbidoul** [comment]
GitHub apparently brought over the discussion we had in https://github.com/sbidoul/peps/pull/1. I assume this is not an issue.

@pfmoore as sponsor will you assign a PEP number ?
@dholth as the first commit is yours it looks like you need to sign the CLA. Or maybe not if commits are squashed before merging.

**Comment by pfmoore** [comment]
I'll wait for the CLA from @dholth just to be on the safe side (although I'm surprised he hasn't already signed it, as he was the author of the wheel PEP).

**Comment by sbidoul** [comment]
The CLA issue is suprising indeed because there are several other commits from Daniel with the exact same e-mail address.

I add to change his email in the document itself because the CI complained that *some authors have more than one email address listed:    Daniel Holth: ['dholth@gmail.com', 'dholth@fastmail.fm']*

I tentatively used number 667 to get a green build.

File: pep-0667.rst
Comment (domdfcoding) [inline pep-0667.rst:79]: Other than here and in the title you've written PEP 517 (i.e. without the hyphen)

File: pep-0667.rst
Comment (domdfcoding) [inline pep-0667.rst:71]:
```suggestion
behave identically as a regular installation. Depending on the way build
backends implement this specification, some minor differences may be visible
```

**Comment by Rosuav** [comment]
PEP number 657 looks to be available; you should be able to grab that. In the (unlikely) event that someone else grabs that number, you'll get a merge conflict, and can bump it to 658.

**Comment by pradyunsg** [comment]
And the title too! :P

**Review by brettcannon — CHANGES_REQUESTED** [review]
Please add Paul in `.github/CODEOWNERS` for this file as the PEP sponsor.

**Comment by brettcannon** [comment]
We also need the CLA to be signed.

File: pep-0657.rst
Comment (pradyunsg) [inline pep-0657.rst:5]:
```suggestion
Discussions-To: https://discuss.python.org/t/8510
```

A cleaner URL.

**Review by pfmoore — CHANGES_REQUESTED** [review]
This needs @dholth to sign the CLA but otherwise it looks OK to me.

**Comment by pfmoore** [comment]
@brettcannon All we're waiting on is the CLA. I don't know whether we're allowed to ignore the CLA requirement and merge this, but other than that, I've approved this...

**Comment by dholth** [comment]
I'll get it taken care of.

**Comment by pradyunsg** [comment]
Given #1950, this likely needs renumbering.

File: .github/CODEOWNERS
Comment (pfmoore) [inline .github/CODEOWNERS:514]: The PEP number 657 has now been taken - can you update this to the next free number? Might be worth waiting until @dholth has got the CLA sorted, so we don't have to keep changing.

File: pep-0657.rst
Comment (pfmoore) [inline pep-0657.rst:1]: PEP number needs changing here too.

**Review by pfmoore — CHANGES_REQUESTED** [review]
PR title and filename will also need changing to reflect the new PEP number.

**Comment by brettcannon** [comment]
More importantly the CLA issue needs to be resolved before fixing the issue number is necessary in case it shifts again.

**Comment by pfmoore** [comment]
I'm not going to be able to keep checking this PR - can @dholth or someone else please explicitly @-ping me when the CLA issue is resolved?

**Comment by pradyunsg** [comment]
@dholth nudge. Any updates on when you might be able to sign the CLA?

**Review by brettcannon — APPROVED** [review]

File: pep-0660.rst
Comment (frostming) [inline pep-0660.rst:145]: A small typo: `achieve`

**Pull request merged**
